Description: FIX libadwaita ignoring light/dark preference

The bug happened due to disablement of light/dark preference in Ubuntu
patches to libadwaita intended to enable Yaru accent color support. This
patch brings back the color scheme support while keeping the accent colors
as well.

libadwaita-1 (1.3.1-1ubuntu2.1) UNRELEASED; urgency=medium

* Non-maintainer upload.
* Force color scheme support on
* Bring back getting color scheme preference from gsettings (legacy
  was not needed on my setup)

Author: Grigorii Khvatskii <gkhvatsk@nd.edu>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/xdg-desktop-portal-gnome/+bug/2013126
Forwarded: not-needed
Last-Update: 2023-04-24

--- libadwaita-1-1.3.1.orig/src/adw-settings.c
+++ libadwaita-1-1.3.1/src/adw-settings.c
@@ -349,6 +349,18 @@ adw_settings_constructed (GObject *objec
     register_impl (self, self->platform_impl, &found_color_scheme, &found_high_contrast);
   }
 
+  //Ubuntu uses gsettings to communicate color scheme preference
+  //No need for legacy here
+  //Portal impl is used above
+  if (!found_color_scheme || !found_high_contrast) {
+    self->gsettings_impl = adw_settings_impl_gsettings_new (!found_color_scheme, !found_high_contrast);
+    register_impl (self, self->gsettings_impl, &found_color_scheme, &found_high_contrast);
+  }
+
+  //Can be replaced by found_color_scheme
+  //But this is a much more reliable way to force the support
+  self->system_supports_color_schemes = TRUE;
+
   init_yaru_accents (self);
 }
 
